python/ray/data/_internal/plan.py

```diff
@@ -610,9 +613,15 @@ def fuse(self, prev: Stage):
 
         block_fn1 = prev.block_fn
         block_fn2 = self.block_fn
+        if prev.target_block_size is not None and self.target_block_size is not None:
+            target_block_size = max(prev.target_block_size, self.target_block_size)
```

Comment:
I believe that taking the max here makes sense, since having to batch within a block is preferable over having far-too-small batch sizes.

Reply:
doesn't it violate the fact that we want to have each batch no more than `batch_size`?

say `ds.map_batches(..., batch_size=100).map_batches(...,)`. If I am the user, I would expect the first map_batches gives me each batch no more than 100 rows, per our API documentation and single `map_batches` behavior.

Reply:
This only affects the ceiling for the cumulative number of rows that we send to a single map task when bundling blocks, it does _not_ affect the ceiling for the size of the batches provided to a UDF, so we won't be violating that guarantee here.